collector/drbd_linux.go

```diff
@@ -0,0 +1,213 @@
+			prometheus.BuildFQName(Namespace, "drbd", name),
+			desc,
+			[]string{"device", "node"}, nil),
+		valueOkay: valueOkay,
+	}
+}
+
+var (
+	drbdNumericalMetrics = map[string]drbdNumericalMetric{
+		"ns": newDrbdNumericalMetric(
+			"network_sent_bytes",
+			"Volume of net data sent to the partner via the network connection.",
+			prometheus.CounterValue,
+			1024),
+		"nr": newDrbdNumericalMetric(
+			"network_received_bytes",
+			"Volume of net data received by the partner via the network connection.",
+			prometheus.CounterValue,
+			1),
+		"dw": newDrbdNumericalMetric(
+			"disk_written_bytes",
+			"Net data written on local hard disk.",
+			prometheus.CounterValue,
+			1024),
+		"dr": newDrbdNumericalMetric(
+			"disk_read_bytes",
+			"Net data read from local hard disk.",
+			prometheus.CounterValue,
+			1024),
+		"al": newDrbdNumericalMetric(
+			"activitylog_writes",
+			"Number of updates of the activity log area of the meta data.",
+			prometheus.CounterValue,
+			1),
+		"bm": newDrbdNumericalMetric(
+			"bitmap_writes",
+			"Number of updates of the bitmap area of the meta data.",
+			prometheus.CounterValue,
+			1),
+		"lo": newDrbdNumericalMetric(
+			"local_pending",
+			"Number of open requests to the local I/O sub-system.",
+			prometheus.GaugeValue,
+			1),
+		"pe": newDrbdNumericalMetric(
+			"remote_pending",
+			"Number of requests sent to the partner, but that have not yet been answered by the latter.",
+			prometheus.GaugeValue,
+			1),
+		"ua": newDrbdNumericalMetric(
+			"remote_unacknowledged",
+			"Number of requests received by the partner via the network connection, but that have not yet been answered.",
+			prometheus.GaugeValue,
+			1),
+		"ap": newDrbdNumericalMetric(
+			"application_pending",
+			"Number of block I/O requests forwarded to DRBD, but not yet answered by DRBD.",
+			prometheus.GaugeValue,
+			1),
+		"ep": newDrbdNumericalMetric(
+			"epochs",
+			"Number of Epochs currently on the fly.",
+			prometheus.GaugeValue,
+			1),
+		"oos": newDrbdNumericalMetric(
+			"out_of_sync_bytes",
+			"Amount of data known to be out of sync.",
+			prometheus.GaugeValue,
+			1024),
+	}
+	drbdStringPairMetrics = map[string]drbdStringPairMetric{
+		"ro": newDrbdStringPairMetric(
+			"node_role_is_primary",
+			"Whether the role of the node is in the primary state.",
+			"Primary"),
+		"ds": newDrbdStringPairMetric(
+			"disk_state_is_up_to_date",
+			"Whether the disk of the node is up to date.",
+			"UpToDate"),
+	}
+
+	drbdConnected = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "drbd", "connected"),
+		"Whether DRBD is connected to the partner.",
+		[]string{"device"}, nil)
+)
+
+type drbdCollector struct{}
+
+func init() {
+	Factories["drbd"] = NewDrbdCollector
+}
+
+func NewDrbdCollector() (Collector, error) {
+	return &drbdCollector{}, nil
+}
+
+func (c *drbdCollector) Update(ch chan<- prometheus.Metric) (err error) {
+	statsFile := procFilePath("drbd")
+	f, err := os.Open(statsFile)
```

Comment:
Just a nit, but the collectors tend to use the full variable name `file` for these type of uses.